code/game/turfs/space/space.dm

```diff
@@ -38,6 +57,30 @@
 
 	return INITIALIZE_HINT_LATELOAD // oh no! we need to switch to being a different kind of turf!
 
+/turf/space/proc/toggle_transit(var/direction)
+	if(edge)
+		return
+
+	if(!direction) //Stopping our transit
+		appearance = SSskybox.dust_cache["[((x + y) ^ ~(x * y) + z) % 25]"]
+	else if(direction & (NORTH|SOUTH)) //Starting transit vertically
+		var/x_shift = SSskybox.phase_shift_by_x[x % (SSskybox.phase_shift_by_x.len - 1) + 1]
+		var/transit_state = ((direction & SOUTH ? world.maxy - y : y) + x_shift) % 15
+		appearance = SSskybox.speedspace_cache["NS_[transit_state]"]
+	else if(direction & (EAST|WEST)) //Starting transit horizontally
+		var/y_shift = SSskybox.phase_shift_by_y[y % (SSskybox.phase_shift_by_y.len - 1) + 1]
+		var/transit_state = ((direction & WEST ? world.maxx - x : x) + y_shift) % 15
+		appearance = SSskybox.speedspace_cache["EW_[transit_state]"]
+
+	for(var/atom/movable/AM in src)
+		if (!AM.simulated || AM.anchored)
+			continue
```

Comment:
I don't get why you got check inverted here, and this continue is also making sure that nothing gets thrown.

```
                for(var/atom/movable/AM in T)
                    if (AM.simulated && !AM.anchored)
                        AM.throw_at(get_step(T, global.reverse_dir[direction]), 5, 1)
                        CHECK_TICK
                    if(istype(AM, /obj/effect/decal)) qdel(AM)
```